docker/ucx-py-cuda11.5.yml

```diff
@@ -7,7 +7,7 @@ dependencies:
   - python=3.9
   - cudatoolkit=11.5
   - setuptools
-  - cython>=0.29.14,<3.0.0a0
+  - cython>=3.0.0
```

Comment:
How did this change here?

Reply:
@pentschev accidentally committed it (it was intentional, just not intended for this PR) and proposed just keeping it: https://github.com/rapidsai/ucx-py/pull/1048#discussion_r1662715558